explore-analyze/scripting/modules-scripting-security.md

```diff
@@ -8,56 +8,55 @@ products:
   - id: elasticsearch
 ---
 
-# Scripting and security [modules-scripting-security]
+# Scripting and security in Painless [modules-scripting-security]
 
-Painless and {{es}} implement layers of security to build a defense in depth strategy for running scripts safely.
+As part of its core design, Painless provides secure scripting capabilities across {{es}}.
 
-Painless uses a fine-grained allowlist. Anything that is not part of the allowlist results in a compilation error. This capability is the first layer of security in a defense in depth strategy for scripting.
+Introduced in [{{es}} 5.0](https://www.elastic.co/blog/painless-a-new-scripting-language) as a replacement for Groovy, Painless is purpose-built for {{es}}, enabling native performance while preventing unauthorized access to system resources.
+   
+By operating in a controlled sandbox environment, Painless ensures that you won’t get compromised when using it. Painless only allows pre-approved operations through fine-grained allowlists. Scripts cannot access file systems, networks, or other system resources that could compromise your cluster while still providing the flexibility you need for search scoring, data processing, and operational automation.
 
-The second layer of security is the [Java Security Manager](https://www.oracle.com/java/technologies/javase/seccodeguide.html). As part of its startup sequence, {{es}} enables the Java Security Manager to limit the actions that portions of the code can take. [Painless](modules-scripting-painless.md) uses the Java Security Manager as an additional layer of defense to prevent scripts from doing things like writing files and listening to sockets.
+## Security Architecture Overview
 
-{{es}} uses [seccomp](https://en.wikipedia.org/wiki/Seccomp) in Linux, [Seatbelt](https://www.chromium.org/developers/design-documents/sandbox/osx-sandboxing-design) in macOS, and [ActiveProcessLimit](https://msdn.microsoft.com/en-us/library/windows/desktop/ms684147) on Windows as additional security layers to prevent {{es}} from forking or running other processes.
+:::{image} /explore-analyze/images/elasticsearch-painless-security-architecture-overview.png
+:alt: Flow chart showing four layers of security between script request and safe script execution
```

Comment:
I'm not sure a flow chart is a good way to present this. Each piece of security is sort of separate from the others in many cases.

Reply:
Good point! I've removed the flow chart graphic, and rephrased the text a bit to de-emphasize the "layers", so as not to suggest that the security features are applied sequentially. I've also removed the bold format from part of the text.